conf/tls/tls_config.go

```diff
@@ -0,0 +1,125 @@
+package tls
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"io/ioutil"
+
+	"github.com/squareup/pranadb/errors"
+
+	"github.com/alecthomas/kong"
+)
+
+// TLSConfig can be used to build a TLS configuration.
+type TLSConfig struct {
+	// nolint: golint
+	CACert     NamedFileContent `help:"PEM-encoded CA certificates to use for ${tls_target}." placeholder:"FILE"`
+	Key        NamedFileContent `help:"Private PEM-encoded key to use for ${tls_target}. Only use if unsetting --key-pair." placeholder:"FILE"`
+	Cert       NamedFileContent `help:"Public PEM-encoded cert to use for ${tls_target}. Only use if unsetting --key-pair." placeholder:"FILE"`
+	ClientAuth string           `help:"Client certificate authentication mode, maps to the tls ClientAuthType"`
+}
+
+var clientAuthTypeMap = map[string]tls.ClientAuthType{
+	"NoClientCert":               tls.NoClientCert,
+	"RequestClientCert":          tls.RequestClientCert,
+	"RequireAnyClientCert":       tls.RequireAnyClientCert,
+	"VerifyClientCertIfGiven":    tls.VerifyClientCertIfGiven,
+	"RequireAndVerifyClientCert": tls.RequireAndVerifyClientCert,
+	"":                           tls.RequireAndVerifyClientCert, // The default value if nothing is provided by the configuration
+}
+
+// NamedFileContent is a custom type that is similar to kong.NamedFileContentFlag but includes an error object.
+type NamedFileContent struct {
```

Comment:
data in config structs should be things like strings and int that directly reflect the contents of the actual config file. They shouldn't contain fields used in processing or for other purposes, e.g. Contents and Error

Reply:
This uses a variation of [Kong CLI's NamedFileContentFlag](https://github.com/alecthomas/kong/blob/a05a0c20ba2d16bce52fbb6e5f59b25a08553880/mapper.go#L776). It implement's [Kong's `MapperValue` interface](https://pkg.go.dev/github.com/alecthomas/kong#MapperValue) to resolve the file's content and return an error if it fails to read the files.

Reply:
We shouldn't use Kong for loading certs from paths. Not everyone will be using Kong and Prana needs to be embeddable and usable without using Kong - Kong should basically just parse args and load the prana config file and that's it.
Take a look at how I've configured TLS for the new HTTP API (WIP), it's super simple:
https://github.com/cashapp/pranadb/blob/http_api/conf/conf.go#L68
https://github.com/cashapp/pranadb/blob/http_api/conf/tls_conf.go
There seems to be a lot of unnecessary complexity in the way things are configured here. 
To read the actual cert is a one liner and can be done when required

Reply:
I can remove using the `MapperValue` interface to resolve the file contents. The entire PranaDB configuration is parsed by Kong library at the moment. The use of Kong is internal to PranaDB. The `prana` and `pranadb` main entry points only receive the `--config` and other flags defined in the configuration:

```
❯ go run ./cmd/pranadb/ --help
Usage: pranadb --config=CONFIG-FLAG

Flags:
  -h, --help                                                           Show context-sensitive help.
      --config=CONFIG-FLAG                                             Path to config file
      --log-format="text"                                              Format to write log lines in
      --log-level="info"                                               Lowest log level that will be emitted
      --log-file="-"                                                   File to direct logs to. If left blank, or '-', logs will go to stdout
      --node-id=INT
      --cluster-id=UINT-64
      --raft-addresses=RAFT-ADDRESSES,...
      --notif-listen-addresses=NOTIF-LISTEN-ADDRESSES,...
      --num-shards=INT
      --replication-factor=INT
      --data-dir=STRING
      --test-server
      --kafka-brokers=BROKER-CONFIGS
      --data-snapshot-entries=INT
      --data-compaction-overhead=INT
      --sequence-snapshot-entries=INT
      --sequence-compaction-overhead=INT
      --locks-snapshot-entries=INT
      --locks-compaction-overhead=INT
      --enable-api-server
      --api-server-listen-addresses=API-SERVER-LISTEN-ADDRESSES,...
      --api-tls-ca-cert=NAMED-FILE-CONTENT                             PEM-encoded CA certificate.
      --api-tls-key=NAMED-FILE-CONTENT                                 Private PEM-encoded key.
      --api-tls-cert=NAMED-FILE-CONTENT                                Public PEM-encoded cert.
      --api-tls-client-auth=STRING                                     Client certificate authentication mode, maps to the tls ClientAuthType
      --enable-source-stats
      --protobuf-descriptor-dir=STRING                                 Directory containing protobuf file descriptor sets that Prana should load to use for decoding Kafka messages. Filenames must end with .bin
      --enable-lifecycle-endpoint
      --life-cycle-listen-address=STRING
      --startup-endpoint-path=STRING
      --ready-endpoint-path=STRING
      --live-endpoint-path=STRING
      --metrics-bind="localhost:9102"                                  Bind address for Prometheus metrics ($METRICS_BIND).
      --enable-metrics
      --enable-failure-injector
      --screen-dragon-log-spam
      --disable-shard-placement-sanity-check
      --raft-rtt-ms=INT
      --raft-election-rtt=INT
      --raft-heartbeat-rtt=INT
      --disable-fsync
      --dd-profiler-types=STRING
      --dd-profiler-host-env-var-name=STRING
      --dd-profiler-port=INT
      --dd-profiler-service-name=STRING
      --dd-profiler-environment-name=STRING
      --dd-profiler-version-name=STRING
      --aggregation-cache-size-rows=INT
      --max-process-batch-size=INT
      --max-forward-write-batch-size=INT
```

Reply:
Just to re-iterate the importance of this. Kong is just something we use in main() in order to load the simple config. main() is just one way to start the server. For example sqlTest does not use it. And if we embed Prana in other apps it will not use it either. That's why it's important we don't have any more complex logic for loading special contents of files in there. Given a conf.Config object with simple file paths it should be possible to start Prana.